mono/mini/interp/transform.c

```diff
@@ -3077,13 +3131,20 @@ interp_emit_sfld_access (TransformData *td, MonoClassField *field, MonoClass *fi
 			}
 		}
 	} else {
-		if (is_load)
+		gpointer field_addr = (char*)mono_vtable_get_static_field_data (vtable) + field->offset;
+		if (is_load) {
+			MonoType *ftype = mono_field_get_type_internal (field);
+			if (ftype->attrs & FIELD_ATTRIBUTE_INIT_ONLY && vtable->initialized) {
```

Comment:
Does this attribute really _guarantee_ us that this field won't ever change again? Could you modify it via reflection?

I guess the answer is "we do it in the JIT too" and CI passes, so it should be fine.

Reply:
I think you could modify it through reflection or maybe some special il assembly, but that is broken and not meant to be supported.